File: JS/ReduzindoUmNumeroZero.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./ReduzindoUmNumeroZero.js')

TEST('Starting ReduzindoUmNumeroZero test...', (t) => {
    //Test 1
    let want = 6
    let ops = 14
    t.assert(OBJ.reduceNumber( ops ) === want, "Expect: 6")
    //Test 2
    want = 4
    ops = 8
    t.assert(OBJ.reduceNumber( ops ) === want, "Expect: 4")

    t.end()
})

File: GO/ReduzindoUmNumeroZero_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestReduzindoUmNumeroZero(t *testing.T) {
	tables := []struct {
		num  int
		want int
	}{
		{14, 6},
		{8, 4},
	}

	for _, table := range tables {
		got := reduceNumber(table.num)

		if got != table.want {
			t.Errorf("Waiting for this %d to %d but the return was this: %d", table.want, table.num, got)
		}
	}
}
